storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2672,7 +2680,12 @@ private ClusterSummary getClusterInfoImpl() throws Exception {
                 summary.set_assigned_memoffheap(resources.getAssignedMemOffHeap());
                 summary.set_assigned_cpu(resources.getAssignedCpu());
             }
-            summary.set_replication_count(getBlobReplicationCount(ConfigUtils.masterStormCodeKey(topoId)));
+            try {
+                summary.set_replication_count(getBlobReplicationCount(ConfigUtils.masterStormCodeKey(topoId)));
+            } catch (KeyNotFoundException e) {
+                // This could fail if a blob gets deleted by mistake.  Don't crash nimbus.
+                LOG.error("Unable to find blob entry", e);
```

Comment:
Does it represent one of blobs for a topology is broken? If then we may need to have specific error message, not general one like this. And if it can't be restored automatically, we may want to guide the way how to fix this situation.

Reply:
@HeartSaVioR  - This change I made first in isolation (when still allowing the blob to be deleted).  This prevents nimbus from crashing, but a nimbus restart will still have issues.  With the other change, this line may no longer be necessary, but I would rather be defensive.  It should not hit unless we have further race conditions the other code check is missing.